projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -62,13 +62,38 @@ public function reregister_menu_items() {
 		$this->add_jetpack_menu();
 		$this->add_gutenberg_menus( $wp_admin );
 
-		// Remove Links Manager menu since its usage is discouraged.
+		// Remove Links Manager menu since its usage is discouraged. https://github.com/Automattic/wp-calypso/issues/51188.
 		// @see https://core.trac.wordpress.org/ticket/21307#comment:73.
-		remove_menu_page( 'link-manager.php' );
+		if ( $this->should_disable_links_manager() ) {
+			remove_menu_page( 'link-manager.php' );
+		}
 
 		ksort( $GLOBALS['menu'] );
 	}
 
+	/**
+	 * Check if Links Manager is being used.
+	 */
+	public function should_disable_links_manager() {
+		// The max ID number of the auto-generated links.
```

Comment:
It might helpful to link out to the source of that number for future reference

Reply:
Agreed. A comment saying `See `/wp-content/mu-plugins/wpcom-wp-install-defaults.php` in WP.com` could be enough.

Reply:
Done